tests/Service/PromotionsService/FindActivePromotionsByEntityGroupedByTypeTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Tests\BBC\ProgrammesPagesService\Service\PromotionsService;
+
+use BBC\ProgrammesPagesService\Domain\Entity\CoreEntity;
+use DateTimeImmutable;
+
+class FindActivePromotionsByEntityGroupedByTypeTest extends AbstractPromotionServiceTest
+{
+    /** @var CoreEntity */
+    private $context;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $dbId = 1;
+        $context = $this->mockEntity('Brand', $dbId);
+        $context->method('getDbAncestryIds')->willReturn([$dbId]);
+
+        $this->context = $context;
+    }
+
+    public function testfindActivePromotionsByEntityGroupedByTypeReturnRightStructure()
```

Comment:
How does this test prove that all promotions aren't just put in the super category?